### PR TITLE
Add article list page and menu item

### PR DESCRIPTION
**Pull request by NathanielBennett**
Following https://github.com/guardian/frontend/pull/8841 and again running inside a 0% a/b test this is allows the user to access a list of previously saved artices. As this is internal only, there's still some styling to to do: 

Also adds a link ( for users in the test only ) to the page in the profile dropdown

saved-for-later

### CODE REVIEW

**Comment by crifmulholland** [comment]
This is great. Very exciting!

File: identity/app/views/profile/savedContent.scala.html [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+            <div>
+                 <div class="saved-content__name"><a href="@article.href">@article.href</a></div>
+                 <div class="saved-content__read">@if(article.read){Read}else{not read}</div>
+            </div>
+            <div class="saved-content__meta u-cf">
+                <button class="saved-content__button"
+                data-link-name="Delete"
+                name="deleteSaveArticle"
+                type="submit"
```
Comment (OliverJAsh) [inline identity/app/views/profile/savedContent.scala.html]: I believe this is the default `type`?

File: identity/app/views/profile/savedContent.scala.html [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                value="@article.shortUrl">
+                Delete
+                </button>
+                <div class="saved-content__saved-at"></i>@article.savedAt</div>
```
Comment (OliverJAsh) [inline identity/app/views/profile/savedContent.scala.html]: Looks like you don't need the `</i>`.

**Comment by phamann** [comment]
:+1: Looks good to me!

**Comment by OliverJAsh** [comment]
:+1:

File: identity/app/controllers/SaveContentController.scala [hunk 3]
```diff
@@ -18,7 +21,10 @@ class SaveContentController @Inject() ( api: IdApiClient,
                                         )
   extends Controller with ExecutionContexts with SafeLogging {
 
-  import authenticatedActions.authAction
+  import authenticatedActions._
```
Comment (rich-nguyen) [inline identity/app/controllers/SaveContentController.scala]: Move this up, and take line 5 and 6 out?

File: identity/app/model/FrontendSavedArticles.scala [hunk 4]
```diff
@@ -2,16 +2,16 @@ package model
 
 import com.gu.identity.model.{SavedArticle, SavedArticles}
 import org.joda.time.DateTime
-import org.joda.time.format.ISODateTimeFormat
+import org.joda.time.format.{DateTimeFormat, ISODateTimeFormat}
 
 /**
```
Comment (rich-nguyen) [inline identity/app/model/FrontendSavedArticles.scala]: Can you remove this auto-comment whilst you're here?

File: identity/app/model/FrontendSavedArticles.scala [hunk 5]
```diff
@@ -2,16 +2,16 @@ package model
...
  * Created by nbennett on 27/03/15.
  */
-class FrontendSavedArticles(version: String, articles: List[SavedArticle]) extends SavedArticles(version, articles) {
+class FrontendSavedArticles(version: String, articles: List[FrontendSavedArticle]) extends SavedArticles(version, articles) {
```
Comment (rich-nguyen) [inline identity/app/model/FrontendSavedArticles.scala]: I'm not sure about the value of `FrontendSavedArticle`, see my comment in the PR

File: identity/app/model/FrontendSavedArticles.scala [hunk 6]
```diff
@@ -2,16 +2,16 @@ package model
...
 
   val fmt = ISODateTimeFormat.dateTimeNoMillis()
 
-  def this() = this(ISODateTimeFormat.dateTimeNoMillis().print(new DateTime()), List[SavedArticle]())
+  def this() = this(ISODateTimeFormat.dateTimeNoMillis().print(new DateTime()), List.empty)
```
Comment (rich-nguyen) [inline identity/app/model/FrontendSavedArticles.scala]: Again, see PR thread

File: identity/app/model/FrontendSavedArticles.scala [hunk 7]
```diff
@@ -26,3 +26,11 @@ class FrontendSavedArticles(version: String, articles: List[SavedArticle]) exten
       case _ => SavedArticles(version, articleToSave :: articles) }
   }
 }
+
+class FrontendSavedArticle(id: String, shortUrl: String, date: DateTime, read: Boolean) extends SavedArticle(id, shortUrl, date, read) {
```
Comment (rich-nguyen) [inline identity/app/model/FrontendSavedArticles.scala]: See PR, think we should remove

**Comment by rich-nguyen** [comment]
I think that you've added `FrontendSavedArticle` to add functionality to the external class, but you can do this using an implicit. That way, you can avoid type casting. Other than that, :+1:

**Pull request merged**